### DISCUSSION TITLE
Getting data from Thanos component via http request

### DISCUSSION THREAD

**Discussion by skandarchahbouni** [thanos-io/thanos#6940] [Questions & Answers]
Hello all,

I am new to Thanos, and I need to retrieve data from a Thanos component (querier or store bucket) programmatically via HTTP requests. Retrieving data programmatically from Prometheus seems straightforward, and I found a helpful guide [here](https://prometheus.io/docs/prometheus/latest/querying/api/). However, I am having difficulty finding equivalent documentation or resources to do a similar thing in Thanos.

For instance, I have a PromQL query, and I need to send this query via an HTTP request (not using the dashboard) to the querier or another Thanos component.

If anyone can provide guidance or assistance, I would greatly appreciate it. Thank you.

**Comment by MichaHoffmann** [comment]
At least thanos querier should expose the same http api ( mostly ). Are you getting empty responses? If so how are things configured?

**Comment by skandarchahbouni** [comment]
No, I haven't tried it yet. However, I searched for information and couldn't find any documentation or details about whether this is possible and which endpoint I have to use. I am also unsure about how to write the PromQL query in the request..........

**Reply by MichaHoffmann** [reply]
I think something like 
```
curl -g 'http://localhost:9090/api/v1/query?' --data-urlencode 'query=up'
```
that points at your thanos querier should work, right?
